### PR TITLE
Api refactoring + log system

### PR DESCRIPTION
**Pull request by hoenirvili**
This commit contains:
- some refactoring throw the winrm api like removing unnecessary var declarations, removing redundant pointers, formatting etc.

### CODE REVIEW

**Comment by hoenirvili** [comment]
Also @masterzen please check this opinion.
https://github.com/hoenirvili/winrm/pull/5#discussion_r65062275

**Comment by masterzen** [comment]
I haven't fully read the `log` commit, but you will need to really separate the logging code from the rest of the refactoring. Right now, this second commit contains some refactoring pertaining to either a different commit of the first one.

I'll read and make more comments on the logging system. At first glance, I didn't want to have any concrete implementation, just an "interface" that does nothing, then the client code would implement one mapping our `Debug()`. `Info()`, etc methods to a concrete implementation using a logger library. In my idea, winrm wouldn't even depend on `log` at all.

I need to think a bit more about this probably :)

**Comment by hoenirvili** [comment]
@masterzen for now I remove the log logic leaving just with the refactoring and some error type. If you want that errorType to be more descriptive and what fields should contain.

**Comment by hoenirvili** [comment]
@masterzen merge?

File: client.go [hunk 1]
```diff
@@ -71,23 +73,23 @@ func newTransport(endpoint *Endpoint) (*http.Transport, error) {
...
 	return certPool, nil
 }
 
-// CreateShell will create a WinRM Shell, which is the prealable for running
-// commands.
-func (client *Client) CreateShell() (*Shell, error) {
-	request := NewOpenShellRequest(client.url, &client.Parameters)
+// CreateShell will create a WinRM Shell,
+// which is the prealable for running commands.
+func (c Client) CreateShell() (*Shell, error) {
```
Comment (masterzen) [inline client.go]: Is that normal that you `c` is not a pointer to a `Client` anymore?

File: client_test.go [hunk 2]
```diff
@@ -12,7 +12,8 @@ import (
 )
 
 func (s *WinRMSuite) TestNewClient(c *C) {
-	client, err := NewClient(&Endpoint{Host: "localhost", Port: 5985}, "Administrator", "v3r1S3cre7")
+	endpoint := NewEndpoint("localhost", 5985, false, false, nil, nil, nil, 0)
```
Comment (masterzen) [inline client_test.go]: I'm not convinced its more clear :)

File: command.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ type commandReader struct {
 type Command struct {
 	client   *Client
 	shell    *Shell
-	ID       string
```
Comment (masterzen) [inline command.go]: This is an incompatible API change.

File: error.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package winrm
+
+import "fmt"
+
+// errWinrm generic error struct
+type errWinrm struct {
```
Comment (masterzen) [inline error.go]: Are we using this error somewhere?
Or is it for future use?

File: response.go [hunk 5]
```diff
@@ -3,61 +3,71 @@ package winrm
...
+func any(node *xmlpath.Node, xpath string) (bool, error) {
 	path, err := xmlpath.CompileWithNamespace(xpath, soap.GetAllNamespaces())
 	if err != nil {
-		return
+		return false, err
 	}
 
-	found = path.Exists(node)
-	return
+	found := path.Exists(node)
```
Comment (masterzen) [inline response.go]: you could just write:
~~~go
return path.Exists(node), nil
~~~

**Comment by masterzen** [comment]
Besides the comments, this looks good.

File: command.go (hunk 3 above)
Comment (hoenirvili) [inline command.go]: I've updated all other ids to be lower case , the ID is exported but we don't need that attribute to be exported.

File: client.go (hunk 1 above)
Comment (hoenirvili) [inline client.go]: mistyped, sorry.

File: error.go (hunk 4 above)
Comment (hoenirvili) [inline error.go]: Just for future use, I could update all errors to be this type instead of generic fmt.Errof().

File: response.go (hunk 5 above)
Comment (hoenirvili) [inline response.go]: Yeah, you are right.

**Comment by hoenirvili** [comment]
Merge?

**Pull request merged**